mlir/include/mlir/Dialect/Rock/IR/RockAttrDefs.td

```diff
@@ -223,14 +223,13 @@ def Rock_GeneralGemmParamsAttr : Rock_Attr<"GeneralGemmParams", [RockTuningParam
 
   let extraClassDeclaration = [{
     void getPerfConfigStr(std::string &perfStr){
-      const Twine paramsConcat =
-        Twine(getBlockSize()) + ","
-      + Twine(getMPerBlock()) + ","
-      + Twine(getNPerBlock()) + ","
-      + Twine(getKPerBlock()) + ","
-      + Twine(getMPerThread()) + ","
-      + Twine(getNPerThread());
-      perfStr = paramsConcat.str();
+      perfStr =
+        std::to_string(getBlockSize()) + ","
```

Comment:
The Twines were making temps on the stack and retaining references to them in the result that was passed to perfStr.  I'm not sure exactly how, but the comments do say that Twine is intended for temporary use.  I made them all std::string because I know how that behaves.

Reply:
That should be "all those twines in parens" then (`to_str()`)

Or changing that `std::string&` to `SmallVectorImpl<char>&` and `to_vector()` on `Twine`.

`Twine` is important to prevent all the `std::string` stuffles.

Reply:
Did go with an "all Twines in parens" then .str() strategy, but also changed the function to return a value instead of update a reference parameter